cirq/circuits/qasm_output.py

```diff
@@ -54,24 +54,26 @@ def from_matrix(mat: np.array) -> 'QasmUGate':
             pre_phase / np.pi,
         )
 
-    def _qasm_(self, qubits: Tuple[ops.Qid, ...], args: 'cirq.QasmArgs') -> str:
+    def _qasm_(self, qubits: Tuple['cirq.Qid', ...],
+               args: 'cirq.QasmArgs') -> str:
         args.validate_version('2.0')
         return args.format(
             'u3({0:half_turns},{1:half_turns},{2:half_turns}) {3};\n',
             self.theta, self.phi, self.lmda, qubits[0])
 
     def __repr__(self) -> str:
-        return 'cirq.QasmUGate({}, {}, {})'.format(self.theta, self.phi,
-                                                   self.lmda)
-
-    def _unitary_(self) -> np.ndarray:
-        # Source: https://arxiv.org/abs/1707.03429 (equation 2)
-        operations = [
-            ops.Rz(self.phi * np.pi),
-            ops.Ry(self.theta * np.pi),
-            ops.Rz(self.lmda * np.pi),
+        return (f'cirq.circuits.qasm_output.QasmUGate('
```

Comment:
Maybe import `QasmUGate`  into top level namespace instead? The name is unlikely to collide and the repr would look a lot better.

Reply:
Will leave it here for now. Can be done in separate change if needed.